server/src/views/frame.js

```diff
@@ -115,14 +64,14 @@ class Snippet extends React.Component {
     }
 
     return <div className="snippet-container">
-      <a href={'#clip=' + encodeURIComponent(clip.id)}>
+      <a href={'?clip=' + encodeURIComponent(clip.id)}>
         { node }
       </a>
       <p>
         <img ref="commentBubble" className="comment-bubble"
           src={ closed ? this.props.staticLink("img/comment-bubble.png") : this.props.staticLink("img/comment-bubble-open.png") }
           onClick={ this.onClickComment.bind(this) } />
-        <a href={'#clip=' + encodeURIComponent(clip.id)}>
+        <a href={'?clip=' + encodeURIComponent(clip.id)}>
```

Comment:
It'll never be more accurate than doing the rendering on the client, where we can use DOM introspection to actually locate the elements.

We need to do client-side analytics anyway, we can never capture enough on the server alone.  Plus if we use pushstate to suppress a reload (which we'd have to do – reloading on clip selection is way too slow) we wouldn't get the server analytics either.